temoto_context_manager/include/temoto_context_manager/env_model_repository.h

```diff
@@ -0,0 +1,199 @@
+#ifndef ENV_MODEL_REPOSITORY_H
+#define ENV_MODEL_REPOSITORY_H
+
+namespace temoto_context_manager 
+{
+namespace emr 
+{
+
+
+
+/**
+ * @brief Abstract base class for payloads
+ * 
+ */
+class PayloadEntry
+{
+private:
+  
+public:
+  std::string type;
+
+  PayloadEntry(std::string type) : type(type) {}
+
+  ~PayloadEntry() {}
+
+  PayloadEntry() {}
+
+  virtual std::string getName() = 0;
+  /**
+   * @brief Get the type of the Payload
+   * 
+   * @return std::string 
+   */
+  std::string getType() {return type;}
```

Comment:
It's a good habit to declare a function as "const" if it will not change the contents of the object itself (std::string getType() const)